vignettes/lifecycle.Rmd

```diff
@@ -18,117 +18,101 @@ knitr::opts_chunk$set(
 library(lifecycle)
 ```
 
-Use lifecycle to document the status of your exported functions and arguments:
+Use lifecycle to document the status of your exported functions and arguments.
+It provides a standard way to describe lifecycle stage in the documentation, and tools to encourage users away from deprecated functions.
 
--   Choose one of the 7 lifecycle stages a function or argument can be in.
-    You can choose from 4 development stages (experimental, maturing, stable, and questioning) and 3 deprecation stages (soft-deprecated, deprecated, and defunct).
+## Stages
 
--   If the function or argument is deprecated, make sure your users know about by calling `deprecate_soft()`, `deprecate_warn()`, or `deprecate_stop()`.
-    These functions try to be informative without being too verbose, with increasing levels of verbosity as the deprecation stage advances.
+The lifecycle stages for functions[^1] are summarised in the figure below.
 
--   Include the relevant lifecycle badge in your documentation.
-
-### Stages
-
-The lifecycle stages for functions and arguments are summarised in the figure below.
-They're designed to closely mirror the lifecycle stages for [packages](https://www.tidyverse.org/lifecycle/).
+[^1]: They can also be applied to individual arguments, or combinations of argument values.
 
 <img src="figures/lifecycle.svg" width="75%"/>
 
-There are two development stages.
-
--   <img src="figures/lifecycle-experimental.svg" alt="Experimental" style="vertical-align:middle"/> This is a new feature that is in the very early stage of development.
-    It is exported so users can start to use it and report feedback, but its interface and/or behaviour is likely to change in the future.
-    It is generally best to avoid depending on experimental features.
-
--   <img src="figures/lifecycle-maturing.svg" alt="Maturing" style="vertical-align:middle"/> The interface and behaviour of a maturing feature has been roughed out, but finer details are likely to change.
-    It still needs more feedback to find the optimal API.
+The default development stage is usually ![Stable](figures/lifecycle-stable.svg){style="vertical-align:middle"}.
+A function is considered stable when the author is happy with its interface and behaviour and major changes are unlikely.
+Breaking changes will occur gradually, through a deprecation process.
 
-The probable end state of most functions is *stable*:
+The exception to the rule is sub-1.0.0 packages where all functions are considered ![Experimental](figures/lifecycle-experimental.svg){style="vertical-align:middle"}.
```

Comment:
I thought that "experimental" means out of warranty? In that case, I don't think this is correct. Most functions in sub-1.0.0 packages (i.e. maturing packages) are in-warranty. I'd say they are maturing.

Reply:
I think strictly speaking that they are out of warranty — but we are careful with breaking changes even for functions  that aren't in warranty. For example, this is why we did all the checking with magrittr — technically I think all the problems we found would be clear violations of the warranty, but there's still some implied contract that we don't make breaking changes without considering the consequences.

I think we have certain obligation as soon as a package is on CRAN, but we don't want to make them as explicit to give us as much freedom as possible to make changes (while still being careful of breaking changes).

Reply:
i.e. we reserve the right to making breaking changes in experimental packages, but really what we consider is (as far as we can tell) actual usage in the wild. Labelling a package as experimental is more about communicating that it is most appropriate to use by people who have some tolerance for risk. 

I think that suggests it's our policy for when a package becomes 1.0.0 really needs to change, because we are willing to provide those guarantees earlier. i.e. purrr is sufficiently popular that we wouldn't make major breaking changes.